### ISSUE TITLE
Learning rate 'Exponential Decay' policy wrong.

### ISSUE THREAD

**Issue by TimZaman** [NVIDIA/DIGITS#662]
Version **v3.2.0-61-g0ce28b9**
I often have issues with the learning rate and the decay. Sometimes it tells me it cannot process 'nan', sometimes it's correct, sometimes it jitters a bit up and down, but most often it does this. I can reproduce this only half of the time. 
Base learning rate: 1.0
Policy: Exponential Decay
Gamma: 0.95
![screen shot 2016-03-29 at 3 16 24 pm](https://cloud.githubusercontent.com/assets/7721540/14109283/8d07aa0e-f5c1-11e5-8e38-7a8fedae70a6.png)

Then this happens:
![screen shot 2016-03-29 at 3 16 11 pm](https://cloud.githubusercontent.com/assets/7721540/14109287/90dd9634-f5c1-11e5-8c0a-2d6637aee321.png)

**Comment by gheinrich** [comment]
Thanks for the bug report. The learning rate seems to be decreasing way too quickly indeed. Are you using Torch or Caffe? I think I can reproduce your issue with Torch.

**Comment by TimZaman** [comment]
Torch indeed. Did not try Caffe.

**Comment by gheinrich** [comment]
@TimZaman do you mind verifying that #663 fixes the issue? Thanks!

**Comment by TimZaman** [comment]
Yes. Merci beaucoup Greg!

**Comment by TimZaman** [comment]
Hmm even after the update, still something fishy here: (500 epochs used)
![screenshot 2016-03-31 15 34 53](https://cloud.githubusercontent.com/assets/7721540/14177361/3953c032-f756-11e5-9dc7-5046c61d8296.png)
![screenshot 2016-03-31 15 34 59](https://cloud.githubusercontent.com/assets/7721540/14177362/396686ea-f756-11e5-9e34-ed5be7f95949.png)

**Comment by gheinrich** [comment]
I just tried running 500 epochs of training. This is what the learning rate looked like:
![lr-exp-decay-500epochs](https://cloud.githubusercontent.com/assets/3889770/14181453/d4cee660-f75e-11e5-8936-342d639d7ac6.png)
I was using this version:
```
$ git describe
v3.2.0-73-gb4dd9c7
````

**Comment by gheinrich** [comment]
@TimZaman were you able to verify that you were using the right version of DIGITS? The exponential decay is working for me for 500 epochs.

**Comment by gheinrich** [comment]
No response and this is working for me on v3.2.0-73-gb4dd9c7 so I am closing the bug.

**Comment by TimZaman** [comment]
i was using the most up to date version for sure. There's something going on with the learning rate for sure, i will try to hunt down the cause next time instead of posting the problem.
